### PR TITLE
Use fetcher pod in upgrade test

### LINKED ISSUE #1892: Backport eventing #4460
**Problem**
This PR use a fetcher pod to pull report in upgrade test instead of replying on external cluster IP address, which is not necessarily available. 

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for?

**Exit Criteria**
The upgrade test works without external IP resolve.

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve?

**Additional context (optional)**
Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by zhongduo**
Update tools.go, run:
./hack/update-deps.sh --upgrade --release v9000.1 && ./hack/update-codegen.sh

Fixes #1892 

## Proposed Changes

- Update dependency to latest
- Apply our change on top of the new upgrade test
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: vendor/knative.dev/eventing/pkg/apis/duck/v1alpha1/subscribable_types.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ var (
 // This is w.r.t https://github.com/knative/eventing/pull/1685#discussion_r314797276
 // Due to change in the API, we support reading of SubscribableTypeStatus#DeprecatedSubscribableStatus in a logical way
 // where we read the V2 value first and if the value is absent then we read the V1 value,
-// Having this function here makes it convenient to read the default value at runtime.
+// Having this function here makes it convinient to read the default value at runtime.
```
Comment (tommyreddad) [inline vendor/knative.dev/eventing/pkg/apis/duck/v1alpha1/subscribable_types.go:120]: lol I wonder how this got through

Comment (zhongduo) [reply vendor/knative.dev/eventing/pkg/apis/duck/v1alpha1/subscribable_types.go]: It is already fixed in the latest eventing.

**Pull request merged**